newsparser.py

```diff
@@ -29,25 +30,69 @@ def getHumanReadableDate(date):
 
 
 class articleParser(HTMLParser):
+  # articleParser parses the raw html from Discourse and generates html that
+  # Telegram likes. During the process it performs the following.
+  #
+  # Convert Tags: Telegram supports a very small subset of html tags. In order
+  #               to provide (some) formatting in the telegram message, parser
+  #               converts tags that telegram doesn't like into ones it does
+  #               or removes them (not the contents) if no conversion exists.
+  # Handle Image: Images are wrapped in <a> tags with href same as img's src.
+  #               (<a><img></a>) In this case parser removes the <img> tag
+  #               completely since Telegram handles <a> tags pretty well with
+  #               previews. (Also telegram doesn't support <img> tags)
+  # Handle Emoji: Emojis are represented as <img> tags but they are not wrapped
+  #               in <a> tags, unlike the previous case. Parser replaces <img>
+  #               tags with 'class="emoji"' with their unicode codepoint.
+
+  # Tags supported by Telegram
+  supported_tags = [
+      'strike', 'i', 'strong', 'b', 'pre', 'ins', 'a', 'code', 'del', 's', 'em',
+      'u'
+  ]
+  # Useful attributes to look for
+  useful_attrs = ['href', 'title', 'class']
+  # html_tag -> telegram_tag mapping for formatting purposes
+  tag_mapping = {'h1': 'b', 'h2': 'b', 'h3': 'b', 'blockquote': 'i'}
 
   def __init__(self):
-    super().__init__()
+    super().__init__(convert_charrefs=False)
     self.reset()
     self.fed = []
 
+  def handle_entityref(self, name):
+    # This is done so that html entities like '&gt;' remain escaped.
+    self.fed.append(f'&{name};')
+
   def handle_starttag(self, tag, attrs):
-    if tag != "img":
+    tag_attrs = {at[0]: at[1] for at in attrs if at[0] in self.useful_attrs}
```

Comment:
why not just create a mapping for all attributes, dropping the need for `useful_attrs`

Reply:
done